src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -0,0 +1,80 @@
+package main.java.com.google.sps.servlets;
+
+import com.google.api.gax.paging.Page;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+
+import java.util.*;
+
+@WebServlet("/comment")
+@MultipartConfig
+public class CommentsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("text/html;");
+    response.getWriter().println("<h1>Show Comments.</h1>");
+    response.getWriter().println("<ul>"); // <ul> --> 'unordered list element'
+    
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+    // Setting an order by time posted
+    Query<Entity> query = Query.newEntityQueryBuilder()
+      .setKind("Comment")
+      .setOrderBy(OrderBy.desc("timestamp"))
+      .build();
+    QueryResults<Entity> results = datastore.run(query);
+
+    // Basicallly stored like(?) a list. This is printing comments.
+    while (results.hasNext()) {
+      Entity entity = results.next();
+      String message = entity.getString("text");
+      response.getWriter().println("<li>" + message + "</li>");
+    }
+    
+    response.getWriter().println("</ul>");
+    response.getWriter().println("<p><a href=\"/\">Back</a></p>"); // tbh a bit confused about this.
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Getting comment input
```

Comment:
+1 Very readable! And it would help to have function-level comments above the function definition to describe what the overall function does. For example:
- for doGet(): Returns HTML to display all comments, sorted by most recent first.
- for doPost(): Creates and stores a new comment. [something about redirect]